src/Mcamara/LaravelLocalization/LaravelLocalizationServiceProvider.php

```diff
@@ -51,8 +51,25 @@ public function register()
      */
     protected function registerBindings()
     {
-        $this->app->singleton(LaravelLocalization::class, function () {
-            return new LaravelLocalization();
+        tap(fn () => new LaravelLocalization(), function ($fn) {
+            // the conditional check below is important
+            // when you do caching routes via `php artisan route:trans:cache` if binding
+            // via `bind` used you will get incorrect serialized translated routes in cache
+            // files and that's why you'll get broken translatable route URLs in UI
+
+            // again, if you don't use translatable routes, you may get rid of this check
+            // and leave only 'bind()' here
+            if ($this->runningInOctane()) {
+                // the 3rd parameter is important to be passed to 'bind()'
+                // otherwise the package's instance will be instantiated every time
+                // you reference it and it won't get proper data for 'serialized translatable routes'
+                // class variable, this will make impossible to use translatable routes properly
+                // but overall the package will still work stable except generating the same URLs
+                // for translatable routes independently of locale
+                $this->app->bind(LaravelLocalization::class, $fn, true);
+            } else {
+                $this->app->singleton(LaravelLocalization::class, $fn);
+            }
```

Comment:
`singleton()` is just a `bind()` with `true` as a third parameter, so this check shouldn't make any difference.

https://github.com/laravel/framework/blob/9.x/src/Illuminate/Container/Container.php#L388